packages/@uppy/tus/src/index.js

```diff
@@ -252,6 +259,24 @@ module.exports = class Tus extends BasePlugin {
         resolve(upload)
       }
 
+      uploadOptions.onShouldRetry = (err, retryAttempt, options) => {
+        if (err?.originalResponse?.getStatus() === 429) {
+          if (!this.requests.isPaused) {
+            const { done, value } = this.#retryDelayIterator.next()
+            if (done) {
+              return false
+            }
+            this.requests.pause(value)
+          }
+          queuedRequest = this.requests.run(queueRequest)
```

Comment:
I might be missing something obvious but where is `queueRequest` being set? I'm also a bit confused where the newly addded `onShouldRetry` is called?

Reply:
`queueRequest` is being set line 306. I tried moving its definition higher but it was triggering ESLint use-before-declare warnings.

`onShouldRetry` is a `tus-js-client` option: https://github.com/tus/tus-js-client/blob/master/docs/api.md#onshouldretry